3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/MailgunAdapter.scala

```diff
@@ -0,0 +1,205 @@
+      case (Some(body), _) if (body.isEmpty)                        => s"${VendorName} event body is empty: nothing to process".failNel
+      case (Some(body), Some(ct))                                   => {
+        val params = toMap(payload.querystring)
+        try {
+          val bodyMap = getBoundary(ct)
+                          .map(parseMultipartForm(body, _))
+                          .getOrElse(toMap(URLEncodedUtils.parse(URI.create("http://localhost/?" + body), "UTF-8").toList))
+          bodyMap.get("event") match {
+            case None            => s"No ${VendorName} event parameter provided: cannot determine event type".failNel
+            case Some(eventType) => {
+              (payloadBodyToEvent(bodyMap)) match {
+                case (Failure(err)) => err.failNel
+                case (Success(event)) => {
+                  lookupSchema(eventType.some, VendorName, EventSchemaMap) match {
+                    case Failure(str)  => str.fail
+                    case Success(schemaUri) => {
+                      NonEmptyList(RawEvent(
+                        api         = payload.api,
+                        parameters  = toUnstructEventParams(TrackerVersion, params, schemaUri, event, "srv"),
+                        contentType = payload.contentType,
+                        source      = payload.source,
+                        context     = payload.context
+                      )).success
+                    }
+                  } 
+                }
+              }
+            }
+          }
+        } catch { 
+          case NonFatal(e) => {
+            val exception = JU.stripInstanceEtc(e.getMessage).orNull
+            s"${VendorName}Adapter could not parse body: [${exception}]".failNel
+          }
+        }
+      }
+    }
+  }
+  /**
+  * Returns the boundary parameter for a message of media type multipart/form-data
+  * (https://www.ietf.org/rfc/rfc2616.txt and https://www.ietf.org/rfc/rfc2046.txt)
+  *
+  * @param contentType Header field of the form "multipart/form-data; boundary=353d603f-eede-4b49-97ac-724fbc54ea3c"
+  * @return boundary Option[String]
+  */
+  private val boundaryRegex = """multipart/form-data.*?boundary=(?:")?([\S ]{0,69})(?: )*(?:")?$""".r
+  private def getBoundary(contentType: String): Option[String] = contentType match {
+    case boundaryRegex(boundaryString) => Some(boundaryString)
+    case _ => None
+  }
+  /**
+   * Rudimentary parsing the form fields of a multipart/form-data into a Map[String, String]
+   * other fields will be discarded (see https://www.ietf.org/rfc/rfc1867.txt and https://www.ietf.org/rfc/rfc2046.txt).
+   * This parser will only take into account part headers of content-disposition type form-data
+   * and only the parameter name e.g.
+   * Content-Disposition: form-data; anything="notllokingintothis"; name="key"
+   *
+   * value
+   *
```

Comment:
empty lines

Reply:
Actually the empty line above "value" is mandated by the RFC mentioned.

Reply:
ah my bad I thought it was some weird comment formatting